packages/contracts-bedrock/scripts/generate-snapshots.ts

```diff
@@ -0,0 +1,136 @@
+import fs from 'fs'
+import path from 'path'
+
+const outdir = process.argv[2] || path.join(__dirname, '..', 'snapshots')
+const forgeArtifactsDir = path.join(__dirname, '..', 'forge-artifacts')
+
+const getAllContracts = (): Array<string> => {
+  const paths = []
+  const readFilesRecursively = (dir: string) => {
+    const files = fs.readdirSync(dir)
+
+    for (const file of files) {
+      const filePath = path.join(dir, file)
+      const fileStat = fs.statSync(filePath)
+
+      if (fileStat.isDirectory()) {
+        readFilesRecursively(filePath)
+      } else {
+        paths.push(filePath)
+      }
+    }
+  }
+  readFilesRecursively(path.join(__dirname, '..', 'src'))
+
+  // Assumes there is a single contract per file
```

Comment:
Hm, I don't think this is currently a safe assumption, since we have two files with multiple contracts:
- `CrossDomainMessenger.sol`
- `Faucet.sol`

If this is an assumption we want to enforce, we should add a semgrep rule (or similar) to enforce 1 contract per file

Reply:
Good catch. I can detect such multi-contract files in the snapshot script. Will fix.

There's another assumption being made on L78, where solc generates multiple artifacts for different compiler versions. We assume that all compiler versions generate the same ABI. And this assumption seems to hold upon manual inspection.

Reply:
> There's another assumption being made on L78, where solc generates multiple artifacts for different compiler versions. We assume that all compiler versions generate the same ABI. And this assumption seems to hold upon manual inspection.

As yep, I missed that one. This assumption should be safe for as long as we're on 0.8.x (but may not necessarily be true when 0.9.x is released, if we even upgrade to it)